Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs

```diff
@@ -131,7 +129,7 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
           return information;
         }
         // Ok no assertion here. Maybe a method?
-        if (node.Position.Line == position.Line && node.Uri == state.Uri) {
+        if (node.Position.Line == position.Line && state.Compilation.Project.IsImplicitProject && node.Uri == state.Compilation.Project.Uri) {
```

Comment:
Why does/should this depend on whether the compilation is for an implicit project?

Reply:
`state.Compilation.Project.Uri` maps to `.dfy` file only for implicit projects. Basically, in this PR, when project mode is on and you're using a project file, all features that depend on the VerificationTree, which is gutter icons and part of hovering, are turned off. This line will change in the future. I don't think we need to leave a todo. I will add tests in a future PR for the behavior that's still missing, and then remove the property `IsImplicitProject` and fix related code.